src/core/job.c

```diff
@@ -99,6 +99,13 @@ Job* job_free(Job *j) {
         assert(!j->subject_list);
         assert(!j->object_list);
 
+        do {
+                Unit *tu = NULL;
+
+                LIST_FOREACH(triggered_by, tu, j->triggered_by)
+                        LIST_REMOVE(triggered_by, j->triggered_by, tu);
```

Comment:
you iterate through the list and remove each item? You might as well just remove the first item... it's much simpler.

Reply:
https://github.com/systemd/systemd/pull/22378